### PR TITLE
Map - Fix effects breaking mid mission

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- title

The map effects are only applied once, but if the map display is destroyed mid mission (loaded savegame, respawn etc.) the effects are never added again.

### CODE REVIEW

File: addons/map/functions/fnc_initMainMap.sqf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include "script_component.hpp"
+
+params ["_display"];
+private _control = _display displayCtrl 51;
```
Comment (Dystopian) [inline addons/map/functions/fnc_initMainMap.sqf]: magic number

File: addons/map/config.cpp [hunk 2]
```diff
@@ -89,6 +89,11 @@ class RscDisplayMainMap {
             onDraw = QUOTE([ctrlParent (_this select 0)] call DFUNC(onDrawMap));
             #include "MapTweaks.hpp"
         };
+        class InitScript: RscText {
+            onLoad = QUOTE(ctrlParent (_this select 0) call (uiNamespace getVariable 'FUNC(initMainMap)'));
```
Comment (mharis001) [inline addons/map/config.cpp]: Use `Extended_DisplayLoad_EventHandlers`, AFAIK `RscDisplayMainMap` uses `BIS_fnc_initDisplay`

File: addons/map/functions/fnc_initMainMap.sqf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}];
+_control ctrlAddEventHandler ["MouseButtonUp", {
+    if ((_this select 1) == 1) then {
+        GVAR(rightMouseButtonLastPos) = [];
+    };
+}];
+
+//get mouse position on map
+_control ctrlAddEventHandler ["MouseMoving", {
+    GVAR(mousePos) = (_this select 0) ctrlMapScreenToWorld [_this select 1, _this select 2];
```
Comment (mharis001) [inline addons/map/functions/fnc_initMainMap.sqf]: Rather than 3x `select`, use `params`?

File: addons/map/functions/fnc_initMainMap.sqf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}];
+_control ctrlAddEventHandler ["MouseHolding", {
+    GVAR(mousePos) = (_this select 0) ctrlMapScreenToWorld [_this select 1, _this select 2];
```
Comment (mharis001) [inline addons/map/functions/fnc_initMainMap.sqf]: Same as above

File: addons/map/functions/fnc_initMainMap.sqf (hunk 3 above)
Comment (commy2) [reply addons/map/functions/fnc_initMainMap.sqf]: True. The orig didn't use it either, but could be updated.

File: addons/map/config.cpp (hunk 2 above)
Comment (commy2) [reply addons/map/config.cpp]: True. It's called `RscDiary` there however, so I should filter all idd != 12 (`IDD_MAIN_MAP`) though.

**Comment by commy2** [comment]
- close #5408

File: addons/map/functions/fnc_initMainMap.sqf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+//map sizes are multiples of 1280
+GVAR(worldSize) = worldSize / 1280;
+GVAR(mousePos) = [0.5, 0.5];
+
+//Allow panning the lastStillPosition while mapShake is active
+GVAR(rightMouseButtonLastPos) = [];
+
+_control ctrlAddEventHandler ["Draw", {_this call FUNC(updateMapEffects)}];
+_control ctrlAddEventHandler ["MouseMoving", {
+    params ["_control", "_x", "_y"];
```
[the remark this answers is not in the document]
Comment (commy2) [reply addons/map/functions/fnc_initMainMap.sqf]: Don't think that's worth it, tbh.

File: addons/map/CfgEventHandlers.hpp [hunk 6]
```diff
@@ -17,3 +17,9 @@ class Extended_PostInit_EventHandlers {
         serverInit = QUOTE(call COMPILE_FILE(XEH_postInitServer));
     };
 };
+
+class Extended_DisplayLoad_EventHandlers {
+    class RscDiary {
+        GVAR(initMainMap) = QUOTE(ctrlParent (_this select 0) call (uiNamespace getVariable 'FUNC(initMainMap)'));
```
Comment (PabstMirror) [inline addons/map/CfgEventHandlers.hpp]:
```
 Error in expression <ctrlParent (_this select 0) call (uiName>
 Error position: <ctrlParent (_this select 0) call (uiName>
 Error ctrlparent: Type Display (dialog), expected Control
```

File: addons/map/XEH_postInitClient.sqf [hunk 7]
```diff
@@ -8,52 +8,6 @@ LOG(MSG_INIT);
 // Calculate the maximum zoom allowed for this map
 call FUNC(determineZoom);
 
-[{
-    if (isNull findDisplay 12) exitWith {};
```
Comment (PabstMirror) [inline addons/map/XEH_postInitClient.sqf]: I think we still want this line or the effects get added to the briefing maps (maps shown before mission starts)
e.g. night mission, before there were no lighting effects on map
I think to filter them out we need to do something like
```
[{
    params ["_display"];
    private _control = _display displayCtrl IDC_MAP;

    if (_display != findDisplay 12) exitWith {};
    ...
}, _this] call CBA_fnc_execNextFrame; // need to delay a frame or findDisplay doesn't work
````

Comment (commy2) [reply addons/map/XEH_postInitClient.sqf]:
```sqf
if (ctrlIDD _display != IDC_MAP) exitWith {};
```
?

Comment (PabstMirror) [reply addons/map/XEH_postInitClient.sqf]: little easier with that command :laughing:, `IDD_MAIN_MAP`

Comment (commy2) [reply addons/map/XEH_postInitClient.sqf]: Done.

**Pull request merged**